devlib/target.py

```diff
@@ -3041,7 +3044,7 @@ def __init__(self,
                                                working_directory=android_working_directory,
                                                executables_directory=android_executables_directory,
                                                connect=False,
-                                               modules=[], # Only use modules with linux target
+                                               modules=[],  # Only use modules with linux target
```

Comment:
That's unrelated to this PR so maybe don't fix it here, but that's a nice antipattern in there: mutable default values. The default value is created when the function is created. Then that specific instance will be used for all function calls where no value is specified:

```
def foo(x, mylist=[]):
    mylist.append(x)
    return mylist

foo(1)
foo(2)
mylist = foo(3)

assert mylist == [1, 2, 3]
```
So the moral here is: never use mutable values as defaults. `None` is not mutable (or it is mutable but has no state to actually mutate, so both views are actually valid), so it makes a nice default. If an empty iterable default was _really_ wanted, `tuple()` would do it.

Reply:
Removed `modules` param since it defaults to `None` already.